File: datasets/python_source-code/Multiple-B1/organic_9.py
# Binary Tree Traversal - organic submission 9
# Recursive pre-order via static methods, plus an assert-based self-test.

class Node:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None


class TreeOps:
    @staticmethod
    def insert(node, key):
        if node is None:
            return Node(key)
        if key < node.key:
            node.left = TreeOps.insert(node.left, key)
        else:
            node.right = TreeOps.insert(node.right, key)
        return node

    @staticmethod
    def preorder(node):
        if node is None:
            return []
        return [node.key] + TreeOps.preorder(node.left) + TreeOps.preorder(node.right)


def _self_test():
    root = None
    for k in [2, 1, 3]:
        root = TreeOps.insert(root, k)
    assert TreeOps.preorder(root) == [2, 1, 3]
    print("self-test passed")


if __name__ == "__main__":
    _self_test()
    root = None
    for k in [50, 30, 70, 20, 40, 60, 80]:
        root = TreeOps.insert(root, k)
    print("Pre-order:", TreeOps.preorder(root))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b1/organic_9.java
// Binary Tree Traversal - organic submission 9
// Recursive pre-order via static methods, plus self-test.

import java.util.ArrayList;
import java.util.List;

class Node {
    int key;
    Node left;
    Node right;

    public Node(int key) {
        this.key = key;
        this.left = null;
        this.right = null;
    }
}

class TreeOps {
    public static Node insert(Node node, int key) {
        if (node == null) {
            return new Node(key);
        }
        if (key < node.key) {
            node.left = insert(node.left, key);
        } else {
            node.right = insert(node.right, key);
        }
        return node;
    }

    public static List<Integer> preorder(Node node) {
        if (node == null) {
            return new ArrayList<>();
        }
        List<Integer> res = new ArrayList<>();
        res.add(node.key);
        res.addAll(preorder(node.left));
        res.addAll(preorder(node.right));
        return res;
    }

    public static void main(String[] args) {
        Node root = null;
        for (int k : new int[]{50, 30, 70, 20, 40, 60, 80}) {
            root = insert(root, k);
        }
        System.out.println("Pre-order: " + preorder(root));
    }
}
